examples/bpi-monolith-test-suite/bpi/bpi.ts

```diff
@@ -0,0 +1,136 @@
+import { Agreement } from "./components/storage/agreement";
+import { BpiSubject } from "./components/identity/bpiSubject";
+import { Invitation } from "./components/workgroup/invitation";
+import { Workgroup } from "./components/workgroup/workgroup";
+import { Workstep } from "./components/workgroup/workstep";
+import { BpiMessage } from "./components/messaging/bpiMessage";
+import { TransactionPoolComponent } from "./components/transactions/transactionPool";
+import { IMessagingComponent } from "./components/messaging/messaging.interface";
+import { IWorkgroupComponent } from "./components/workgroup/workgroup.interface";
+import { IIdentityComponent } from "./components/identity/identity.interface";
+import { ITransactionPoolComponent } from "./components/transactions/transactionPool.interface";
+import { IVirtualStateMachineComponent } from "./components/vsm/vsm.interface";
+import { VirtualStateMachine } from "./components/vsm/vsm";
+import { IStorageComponent } from "./components/storage/storage.interface";
+import { StorageComponent } from "./components/storage/storage";
+
+export class BPI {
+    identityComponent: IIdentityComponent;
+    messagingComponent: IMessagingComponent;
+    workgroupComponent: IWorkgroupComponent;
+    transactionPoolComponent: ITransactionPoolComponent;
+    vsmComponent: IVirtualStateMachineComponent;
+    storageComponent: IStorageComponent;
+
+    constructor(
+        id: string,
+        name: string,
+        productIds: string[],
```

Comment:
Recommend to not hard code a name here but rather call it `bpiStateObject` and define it simply as an array. That gives you much more flexibility if you want to extend later. But fine for now @ognjenkurtic

Reply:
We intentionally wanted to start with specific use-case before abstracting away at a later stage. Your comment here and in the rest of the places makes perfect sense, but lets tackle this generalization as part of upcoming, smaller tasks [here](https://github.com/eea-oasis/baseline/projects/2)